nomad/structs/acl.go

```diff
@@ -517,3 +555,227 @@ type ACLRoleByNameResponse struct {
 	ACLRole *ACLRole
 	QueryMeta
 }
+
```

Comment:
Can we add some tests for the functionality from this point on?

Reply:
added.